server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/qualitymodel/ReliabilityAndSecurityRatingMeasuresVisitor.java

```diff
@@ -181,35 +180,43 @@ private static void addToParent(Path<Counter> path) {
     }
   }
 
+  private static Measure createRatingMeasure(Rating rating) {
+    return newMeasureBuilder().create(rating.getIndex(), rating.name());
+  }
+
   static final class Counter {
-    private RatingVariationValue reliabilityRating = new RatingVariationValue();
-    private RatingVariationValue securityRating = new RatingVariationValue();
-    private RatingVariationValue.Array newSecurityRating = new RatingVariationValue.Array();
+    private Map<String, RatingVariationValue> ratingValueByMetric = ImmutableMap.of(
```

Comment:
remark: I have the feeling that this type of Map will be associated with behavior tomorrow